### DISCUSSION TITLE
Getting 404 Not Found when clicking ticket links inserted into outbound emails

### DISCUSSION THREAD

**Discussion by trainor1** [django-helpdesk/django-helpdesk#1115] [General]
Hi All,

I am getting 404 Not Found errors when clicking through on ticket links inserted into outbound emails.  My app is using HTTPS.  The links use HTTP.  This might be solved by an Apache forwarding rule.  Unfortunately, our University email is re-writing the URL strings to make use of https://urldefense.com/.  So, the links are breaking at that indirect step.

How can I get Django Helpdesk to use HTTPS when creating the ticket links inserted into outbound emails.

Thanks for your help.

Best,

Kevin

**Comment by Benbb96** [comment]
I think you should be able to control that in the admin module, editing the Site. You have a URL that I believe is used when generating links in mails.

**Comment by trainor1** [comment]
Hi Ben,

Thanks.

This is an interesting idea that I have considered already.  In the documentation for the sites framework, the Site model class has a filed named domain.  I believe that is the field that we are exposing in the admin app.  My current value is just my domain name:  trainorhelpdesk.ligent.net.  The Django documentation for this field gives the example value:  www.example.com.  I could try this.  But, I am doubtful that it supports an expression that includes https://.

Best,

Kevin

**Comment by brucegibbins** [comment]
try the following in your settings file.

HELPDESK_USE_HTTPS_IN_EMAIL_LINK = True

**Reply by trainor1** [reply]
Hi Bruce,

This setting solved my problem.  The system is now inserting HTTPS links into outbound email.  I probably would not have discovered this problem if I didn't have an error in the forwarding rules on my Apache HTTP server.  I just got that fixed too.  I feel more comfortable serving all HTTPS links.  So, I really appreciate you having sent this settings solution.

Thanks of your help.

Best,

Kevin
